tests/utils/test_fmrisim_real_time.py

```diff
@@ -0,0 +1,172 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""fmrisim real-time generator test script
+
+ Authors: Cameron Ellis (Princeton) 2020
+"""
+import numpy as np
+from brainiak.utils import fmrisim_real_time_generator as gen
+import pytest
+import os
+import nibabel as nib
+import time
+import glob
+import copy
+from pkg_resources import resource_stream
+
+# Test that it crashes without inputs
+with pytest.raises(TypeError):
+    gen.generate_data()
+
+data_dict = {}
+data_dict['ROI_A_file'] = resource_stream(__name__, "ROI_A.nii.gz")
+data_dict['ROI_B_file'] = resource_stream(__name__, "ROI_B.nii.gz")
+data_dict['template_path'] = resource_stream(__name__, "sub_template.nii.gz")
+data_dict['noise_dict_file'] = resource_stream(__name__, "sub_noise_dict.txt")
+data_dict['numTRs'] = 30
+data_dict['event_duration'] = 2
+data_dict['scale_percentage'] = 1
+data_dict['different_ROIs'] = True
+data_dict['multivariate_pattern'] = False
+data_dict['save_dicom'] = False
+data_dict['save_realtime'] = False
+data_dict['trDuration'] = 2
+data_dict['isi'] = 4
+data_dict['burn_in'] = 6
+
+
+# Run default test
+def test_default(outputDir=tmp_path, dd=data_dict):
```

Comment:
To use the `tmp_path` fixture, the test function should be declared:
```python
def test_default(tmp_path, dd=data_dict):
```
Fixture work based on argument names, not values:
https://docs.pytest.org/en/latest/fixture.html#fixtures-as-function-arguments

Also, you don't need to clean up `tmp_path`. It is a temporary directory created exclusively for this function:
https://docs.pytest.org/en/latest/tmpdir.html#the-tmp-path-fixture